src/main/java/de/tum/in/www1/artemis/service/connectors/localvc/LocalVCPersonalAccessTokenManagementService.java

```diff
@@ -0,0 +1,38 @@
+package de.tum.in.www1.artemis.service.connectors.localvc;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_LOCALVC;
+
+import java.security.SecureRandom;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+@Service
+@Profile(PROFILE_LOCALVC)
+public class LocalVCPersonalAccessTokenManagementService {
+
+    private static final Logger log = LoggerFactory.getLogger(LocalVCPersonalAccessTokenManagementService.class);
+
+    private static final String TOKEN_PREFIX = "vcpat-";
+
+    private static final int RANDOM_STRING_LENGTH = 44;
+
+    public static final int VCS_ACCESS_TOKEN_LENGTH = TOKEN_PREFIX.length() + RANDOM_STRING_LENGTH; // can be at most 50; database stores token as varchar(50)
```

Comment:
Can we define `RANDOM_STRING_LENGTH` by 50 - `TOKEN_PREFIX`.length() to make sure we are in the db limits?

Reply:
That's a good suggestion, I flipped the assignment.